ibm/service/iampolicy/resource_ibm_iam_policy_template.go

```diff
@@ -0,0 +1,498 @@
+// Copyright IBM Corp. 2023 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package iampolicy
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/platform-services-go-sdk/iampolicymanagementv1"
+)
+
+func ResourceIBMIAMPolicyTemplate() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMIAMPolicyTemplateCreate,
+		ReadContext:   resourceIBMIAMPolicyTemplateVersionRead,
+		UpdateContext: resourceIBMIAMPolicyTemplateUpdate,
+		DeleteContext: resourceIBMIAMPolicyTemplateVersionDelete,
+		Exists:        resourceIBMIAMPolicyTemplateVersionExists,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_iam_policy_template", "name"),
+				Description:  "name of template.",
+			},
+			"account_id": {
```

Comment:
Can we remove this accoutn_id as required and get the value from logged in user context in res/ds
```

userDetails, err := meta.(conns.ClientSession).BluemixUserDetails()
	accountID := userDetails.UserAccount
	createAccountSettingsTemplateOptions.SetAccountID(accountID)
	```
	To be in sync with all other resources